### DISCUSSION TITLE
Which function to call in order to disconnect a client?

### DISCUSSION THREAD

**Discussion by nctnico** [any1/neatvnc#51] [Q&A]
Hello,
I have inherited some code which uses neatvnc to implement a VNC server. What I need to add is a limit to the number of connected clients. So when a new client arrives, I want to disconnect the oldest connection. I have looked through the header file and browsed a bit through the code but it is completely unclear to me which function from the API I have to call in order to drop a client gracefully (which then also calls the client_cleanup callback to gracefully close the connection). I want to do drop the connection from within the new_client callback.

Thanks in advance!

**Comment by any1** [comment]
There is no API function to do that. It can be added; should be easy.

**Reply by nctnico** [reply]
It would be great if you can add such a function!

**Reply by any1** [reply]
Well, I'm rather busy these days. PRs are welcome.

**Reply by nctnico** [reply]
I can probably do it myself but I'd need some hints on which internal functions to call as the internal structure of neatvnc isn't clear to me. I have to avoid a long trial & error path

**Reply by nctnico** [reply]
OK, I have done a bit of hacking and came up with something that seems to work. I've attached a patch.
[nvnc_close_client.patch.txt](https://github.com/any1/neatvnc/files/6488359/nvnc_close_client.patch.txt)
